src/Authenticator/JwtAuthenticator.php

```diff
@@ -166,6 +168,24 @@ protected function decodeToken(string $token): ?object
             );
         }
 
+        $jsonWebKeySet = $this->getConfig('jsonWebKeySet');
+        if ($jsonWebKeySet) {
+            $keySet = JWK::parseKeySet($jsonWebKeySet);
+            /*
+             * TODO Converting Keys to Key Objects is no longer needed in firebase/php-jwt ^6.0
+             * @link https://github.com/firebase/php-jwt/pull/376/files#diff-374f5998b3c572d86be0e79432aac3de362c79e8fb146b9ce422dc2388cdc5daR50
+             */
+            $keyAlgorithms = array_combine(array_column($jsonWebKeySet['keys'], 'kid'), array_column($jsonWebKeySet['keys'], 'alg'));
+            array_walk($keySet, function(&$keyMaterial, $k) use ($keyAlgorithms){
+                $keyMaterial = new Key($keyMaterial, $keyAlgorithms[$k]);
+            });
```

Comment:
Doing this has been suggested here:

https://github.com/firebase/php-jwt/pull/365#issuecomment-1001227542

It is no longer needed when firebase/php-jwt ^6.0 is needed
[
src/JWK.php](https://github.com/firebase/php-jwt/pull/376/files#diff-374f5998b3c572d86be0e79432aac3de362c79e8fb146b9ce422dc2388cdc5daR50)